promsonnet/prom.libsonnet

```diff
@@ -22,24 +39,22 @@
         name: name,
         rules_map:: {},
         rules_order:: [],
+        patches:: [],
         local rules_map = self.rules_map,
         local rules_order = self.rules_order,
-        rules: [rules_map[rule] for rule in rules_order],
       },
 
-      rule: {
-        newAlert(name, rule):: {
-          rules_map+:: {
-            [name]: rule { alert: name },
-          },
-          rules_order+:: [name],
+      newAlertRule(name, rule):: {
```

Comment:
I liked `newAlert` and `newRecording` better than `newAlertRule` and `newRecordingRule`. The latter appears as stutter to me.

Reply:
Perhaps change `newRecording` to `newRecord` to be even more consistent.

Reply:
Ha, I found the opposite true and suggested 'unnesting', just semantics, right?

Reply:
Yes, I saw that. It certainly a matter of style. I just wanted to state that I liked the previous way (`rule.NewAlert` and `rule.NewRecord`) better. I leave it to @malcolmholmes to be the tie breaker.